lib/format_parser.rb

```diff
@@ -255,7 +255,11 @@ def self.parsers_for(desired_natures, desired_formats, filename_hint = nil)
     # Order the parsers according to their priority value. The ones having a lower
     # value will sort higher and will be applied sooner
     parsers_in_order_of_priority = parsers.to_a.sort do |parser_a, parser_b|
-      @parser_priorities[parser_a] <=> @parser_priorities[parser_b]
+      if @parser_priorities[parser_a] != @parser_priorities[parser_b]
+        @parser_priorities[parser_a] <=> @parser_priorities[parser_b]
+      else
+        parser_a.class.name <=> parser_b.class.name
```

Comment:
We really need to explain _why_ we are doing this in a comment

Reply:
I added a comment, could you check it pls?